### PR TITLE
Add namespace to all files header + implementation

### PR DESCRIPTION
**Pull request by david-alvarez-rosa**
All files have been 'namespaced'.

Testing
-------

Build succeeds and executable run as expected.

### CODE REVIEW

File: include/qfm/asset/asset.hpp [hunk 1]
```diff
@@ -18,3 +21,6 @@ class Asset {
   AssetType type_;
   AssetTraitSet traits_;
 };
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset.hpp:26]: Here the namespaces should be `asset` and `qfm`.

File: include/qfm/asset/asset_expiration.hpp [hunk 2]
```diff
@@ -26,3 +30,6 @@ class AssetExpiration {
  private:
   int64_t expiration_;
 };
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset_expiration.hpp:35]: Same here.

File: include/qfm/asset/asset_strike_price.hpp [hunk 3]
```diff
@@ -19,4 +23,7 @@ class AssetStrikePrice {
 
  private:
   double strike_price_;
-};
\ No newline at end of file
+};
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset_strike_price.hpp:29]: Same here.

File: include/qfm/asset/asset_ticker.hpp [hunk 4]
```diff
@@ -16,4 +20,7 @@ class AssetTicker {
 
  private:
   std::string ticker_;
-};
\ No newline at end of file
+};
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset_ticker.hpp:26]: Same here.

File: include/qfm/asset/asset_trait.hpp [hunk 5]
```diff
@@ -29,3 +32,6 @@ class AssetTrait {
   std::string key_;
   std::string value_;
 };
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset_trait.hpp:37]: Same here.

File: include/qfm/asset/asset_trait_set.hpp [hunk 6]
```diff
@@ -37,3 +41,6 @@ class AssetTraitSet {
  private:
   std::unordered_map<std::string, AssetTrait> traits_;
 };
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset_trait_set.hpp:46]: Same here.

File: include/qfm/asset/asset_type.hpp [hunk 7]
```diff
@@ -23,3 +26,6 @@ class AssetType {
  private:
   std::string type_;
 };
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/asset_type.hpp:31]: Same here.

File: include/qfm/asset/option.hpp [hunk 8]
```diff
@@ -17,4 +20,7 @@ class Option : public Asset {
   std::string GetUnderlying() const noexcept;
   AssetStrikePrice GetStrikePrice() const noexcept;
   AssetExpiration GetExpirationDate() const noexcept;
-};
\ No newline at end of file
+};
+
+}  // namespace trait
+}  // namespace asset
```
Comment (waifod) [inline include/qfm/asset/option.hpp:26]: Same here.

**Review by waifod — APPROVED** [review]
Approving, but we need to follow up on a couple of comments. I will take care of them myself tomorrow if you don't do it.

Thanks for the work!

**Comment by waifod** [comment]
Resolving my previous comments since they were addressed in the PR dealing with the formatting.

**Comment by david-alvarez-rosa** [comment]
Thank you for the detailed review! I didn't notice those.

And good to know that `clang-format` takes care of those *automagically*.

**Pull request merged**